src/backend/azure.rs

```diff
@@ -687,6 +697,16 @@ impl StorageBackend for AzureBlobStorageBackend {
             url = url.as_str()
         );
 
+        // Azure doesn't support conditional requests for `Put Block`.
+        // Therefore, we must issue a HEAD request for the blob if not overwriting.
+        // See: https://learn.microsoft.com/en-us/rest/api/storageservices/specifying-conditional-headers-for-blob-service-operations
```

Comment:
Huh, do you know what the deal is with these `x-ms-source` headers?
<img width="871" height="395" alt="image" src="https://github.com/user-attachments/assets/1680232a-7e31-42eb-90a1-9b5c0050e4c6" />